control.lua

```diff
@@ -163,7 +166,15 @@ local function onAirFiltering()
             if (entity.is_connected_to_electric_network() and
                 ((entity.energy / entity.prototype.max_energy_usage) > 0.65)) then
                 local amount = entity.surface.get_pollution(entity.position)
-                if (amount > 0) then
+                if not minDiffuse then
+                    minDiffuse = game.map_settings.pollution.min_to_diffuse
+                end
+                if (not entity.active) and (amount > minDiffuse * 0.75) then
+                    entity.active = true
+                end
+                if (amount < minDiffuse * 0.25) and entity.active then
+                    entity.active = false
+                elseif (amount > 0) then
```

Comment:
This logic is to prevent the power drain?

Reply:
no, this logics does NOT prevent energy drain. Factorio eats drain electrical energy even for deactivated buildings.

The purpose is clearly described in the mod discussion thread. I'll repeat here in other words:
deactivating effectively prevents filters wasting when there is very low amount of pollution units in chunk. Clean filters and usage_energy are NOT WASTED anymore when there is really no pollution in chunk to decrease with negative emission.

Adding 0.05 of actual PUs value does NOT prevent this trouble as your script is doing it every tick, because 60-200 ticks is enough to get 20 units of "pollution gas" even when there is negligible PUs amount (for example 2.0 PUs). A lot of filters and energy is completely wasted.
Now with this fix air filter building goes into deactivated state, waits for some seconds till pollution level rises and activates again to decrease pollution.
PUs value is got from game map settings - it's a key value in factorio pollution mechanics. Unfortunately Wube made it hidden for users, but it is a key in understanding why pollution dissipates sow slow when its level is low and why pollution doesn't migrate to adjacent chunks. Air filter with this check doesn't work when it doesn't help - when pollution will not diffuse in direction of nests anyway.
I love how it works now. The only questionable thing is should script check entity.active property before changing it. I added it in check, because I thought that maybe setting entity.active = false can consume UPS even when entity.active is already false. Here I maybe wrong. And if this field check to be deleted, then code can be optimized further - to not check for positive _amount_ value and skip adding polllution gas automatically.